src/backend/data/post.js

```diff
@@ -29,8 +22,16 @@ class Post {
     this.id = hash(guid);
     this.title = title;
     this.html = html;
-    this.published = datePublished ? toDate(datePublished) : new Date();
-    this.updated = dateUpdated ? toDate(dateUpdated) : new Date();
+    try {
+      this.published = new Date(datePublished);
+    } catch (error) {
+      throw new Error(`post has invalid publication date : ${datePublished}'`);
+    }
+    try {
+      this.updated = new Date(dateUpdated);
+    } catch (error) {
+      throw new Error(`post has invalid date of last update: ${dateUpdated}'`);
+    }
```

Comment:
Maybe you could have `try and catch` inside of `toDate` and throw there. Also, `toDate` checks if what's passed is already an instance of `Date`, I think it'd be useful to keep it.

Reply:
Also, what criteria does `Date` follow to throw? I looked around to see how people check for invalid dates and I found some [interesting suggestions](https://stackoverflow.com/a/1353711/8628446).

Reply:
Made some changes in https://github.com/Seneca-CDOT/telescope/pull/823/commits/24a4fd4fdf0f8c7ce990b8f0ff024912e0f3b96f, let me know what you think!

Reply:
Apologies for the delay. Made the requested changes in https://github.com/Seneca-CDOT/telescope/pull/823/commits/225ef1ac7607100c12f410c1a4c00f14174927ec, please let me know what you think!